static/js/workspace_detail.js

```diff
@@ -0,0 +1,27 @@
+const updateStatuses = (url) => {
```

Comment:
I am well out-of-date with current browser support for various constructions used here. Can you enlighten me?

Reply:
Sure!

[`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const): block scoped variable which can't be changed.

[arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions): a new way to write functions in ES6.  I prefer how concise they are.  The declaration here is equivalent to `updateStatuses(url) { // do stuff };`.

[`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API): browser native, promise-based, function to fetch resources.  Syntax is similar to jQuery's `ajax()` function.

The for loop is the same as Python's `for k, v in some_dict.items():`.  Specifically the assignment is a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) in JS terminology.

Does that cover everything?

Reply:
Thanks, that is useful, but when I said *current browser support* I meant, what will this break in?!

Reply:
ah, misunderstood!

Nothing 🙂

Everything I've used here has been well supported on all major browsers for years.

The only outlier is IE11 but IIRC that's been superseeded by Edge for some time now?

Reply:
IE10 is still common in the NHS...